### DISCUSSION TITLE
When moving to a new line, display the entire line if its possible.

### DISCUSSION THREAD

**Discussion by pathwave** [helix-editor/helix#5926] [Ideas]
When you navigate to the end of a line that is outside the current view, the view is moved so that you can see the cursor.
But when you then navigate back to a shorter line, the view is only moved back enough to see the cursor (+ the offset).

Here I moved up from the end of line 54 to line 53:

![image](https://user-images.githubusercontent.com/114359560/215315412-43a43949-351f-4d5b-a4ef-7d7ec9eb7016.png)

And there is not much information for me to see what line 53 contains.

Instead I think that it would be better to display the entire line if its content (+offset) fits inside the view, otherwise display as much of the line while including the cursor (+offset) as possible.

**Comment by pascalkuthe** [comment]
If this feature is implemented it should be based on #5420 as that PR completly replaces the code that handles view positioning code.

However I am not sure if this behaviour is even desirable. It would not only be a bit hard to implement/make the positioning code slower. Furthermore I think the view jumping around while moving would be quite distracting, no other editor handles it this way either AFAIK. If you always want to retain the full context of the line an easier solution would be to enable softwrap omce #5420 lands.
